models/fluid_body/gw_constituent.py

```diff
@@ -0,0 +1,158 @@
+        on_delete=models.SET_NULL, verbose_name='gwConstituentRelationType',
+        help_text="Specific type of relation between fluid body components, "
+                  "e.g. coating, constitution, aggregation, containment.")
+    gw_constitution_relation_mechanism = models.ForeignKey(
+        MechanismType, null=True, blank=True,
+        on_delete=models.SET_NULL, verbose_name='gwConstitutionRelationMechanism',
+        help_text="Mechanisms by which materials (of various states) come into a relationship, "
+                  "e.g. sorption, precipitation, digestion, excretion, etc.")
+
+
+class GWMaterialConstituent(models.Model):
+    """
+    7.6.27 GW_MaterialConstituent
+    Suspended or colloidal material in a fluid body, e.g., sediment.
+
+    """
+
+    gw_material = models.ForeignKey(
+        MaterialType, null=True, blank=True,
+        on_delete=models.SET_NULL, verbose_name='gwMaterial',
+        help_text="Name of the suspended or colloid material in the fluid body, e.g. a lithology or mineral name.")
+
+    def __str__(self):
+        return self.gw_material
+
+
+class GWChemicalConstituent(models.Model):
+    """
+    7.6.10 GW_ChemicalConstituent
+    Characterization of the chemical composition of the fluid body, both natural and man-made.
+    """
+
+    gw_chemical = models.ForeignKey(
+        ChemicalType, null=True, blank=True,
+        on_delete=models.SET_NULL, verbose_name='gwChemical',
+        help_text="Chemical component type, e.g. arsenic.")
+
+    def __str__(self):
+        return self.gw_chemical
+
+
+class GWBiologicConstituent(models.Model):
+    """
+    7.6.9 GW_BiologicConstituent
+    Characterization of the biological composition of the fluid body, both natural and man-made.
+
+    """
+
+    gw_organism = models.ForeignKey(
+        OrganismType, null=True, blank=True,
+        on_delete=models.SET_NULL, verbose_name='gwOrganism',
+        help_text="Biological species.")
+    gw_state = models.ForeignKey(
+        StateType, null=True, blank=True,
+        on_delete=models.SET_NULL, verbose_name='gwState',
+        help_text="Organisms are always solids.")
+
+    def __str__(self):
+        return self.gw_organism
+
+
+class GWConstituent(models.Model):
+    """
+    7.6.12 GW_Constituent
+    General (abstract) entity denoting a material, chemical or biological constituent of a fluid body.
+    """
+
+    gw_concentration = models.OneToOneField(
+        Quantity, null=True, blank=True,
+        on_delete=models.SET_NULL,
+        verbose_name="gwConcentration",
+        help_text="The concentration (with uom) of the constituent in the fluid body.",
+        related_name='gw_concentration'
+    )
+    gw_state = models.ForeignKey(
+        StateType, null=True, blank=True,
+        on_delete=models.SET_NULL, verbose_name='gwState',
+        help_text="The physical state of the constituent, i.e. solid, liquid, or gas.")
+    gw_constituent_relation = models.ManyToManyField(
```

Comment:
From the diagram, it is not ManyToMany
but GW_Constituent has field gwConstitutedOf
which is
gwConstitutedOf = GWConstituent + GWConstituentRelation

so we can create class 
GWConstituentOfRelation which has : GWConstituent + GWConstituentRelation

and on GWConstituent, it has gwConstitutedOf, which is foreign key of GWConstituentOfRelation